hw/ip/prim/doc/prim_flash.md

```diff
@@ -1,157 +1,159 @@
-  {name: 'done_o',               wave: '0............10..'},
-  {name: 'erase_suspend_done_o', wave: '0............10..'},
-]}
-{{< /wavejson >}}
+---
+title: "Primitive Component: Flash Wrapper"
+---
+
+# Overview
+`prim_flash` is a wrapper interface for technology specific flash modules.
+
+As the exact details of each technology can be different, this document mainly describes the interface requirements and their functions.
+The wrapper however does assume that all page sizes are the same (they cannot be different between data and info partitions, or different types of info partitions).
+
+## Parameters
+
+Name           | type   | Description
+---------------|--------|----------------------------------------------------------
+NumBanks       | int    | Number of flash banks.  Flash banks are assumed to be identical, asymmetric flash banks are not supported
+InfosPerBank   | int    | Maximum number of info pages in the info partition.  Since info partitions can have multiple types, this is max among all types.
+InfoTypes      | int    | The number of info partition types, this number can be 1~N.
+InfoTypesWidth | int    | The number of bits needed to represent the info types.
+PagesPerBank   | int    | The number of pages per bank for data partition.
+WordsPerPage   | int    | The number of words per page per bank for both information and data partition.
+DataWidth      | int    | The full data width of a flash word (inclusive of metadata)
+MetaDataWidth  | int    | The metadata width of a flash word
+TestModeWidth  | int    | The number of test modes for a bank of flash
+
+
+## Signal Interfaces
+
+### Overall Interface Signals
+Name                    | In/Out | Description
+------------------------|--------|---------------------------------
+clk_i                   | input  | Clock input
+rst_n_i                 | input  | Reset input
+flash_req_i             | input  | Inputs from flash protocol and physical controllers
+flash_rsp_o             | output | Outputs to flash protocol and physical controllers
+prog_type_avail_o       | output | Available program types in this flash wrapper: Currently there are only two types, program normal and program repair
+init_busy_o             | output | The flash wrapper is undergoing initialization
+tck_i                   | input  | jtag tck
+tdi_i                   | input  | jtag tdi
+tms_i                   | input  | jtag tms
+tdo_o                   | output | jtag tdo
+scanmode_i              | input  | dft scanmode input
+scan_rst_n_i            | input  | dft scanmode reset
+flash_power_ready_h_i   | input  | flash power is ready (high voltage connection)
+flash_power_down_h_i    | input  | flash wrapper is powering down (high voltage connection)
+flash_test_mode_a_i     | input  | flash test mode values (analog connection)
+flash_test_voltage_h_i  | input  | flash test mode voltage (high voltage connection)
+tl_ul                   | inout  | TL_UL  interface for rd/wr registers access

```

Comment:
it might be better to split this into `tl_i` and `tl_o` separately.

Reply:
sure, you are right